store/snap_remote_repo.go

```diff
@@ -241,6 +245,21 @@ func (s *SnapUbuntuStoreRepository) configureStoreReq(req *http.Request, accept
 	s.applyUbuntuStoreHeaders(req, accept)
 }
 
+// read all the available metadata from the store response and cache
+func (s *SnapUbuntuStoreRepository) readStoreResponse(resp *http.Response) {
+	suggestedCurrency := resp.Header.Get("X-Suggested-Currency")
+	if suggestedCurrency == "" {
```

Comment:
That logic means we'll replace a known currency by a default value. I think we should do this instead:
```
suggestedCurrency := ... from header ...

s.mu.Lock()
defer s.mu.Unlock()
if suggestedCurrency != "" {
        s.suggestedCurrency = suggestedCurrency
}
```

We'll handle defaults when getting it.